src/puppetlabs/trapperkeeper/internal.clj

```diff
@@ -448,20 +475,24 @@
         this)
       (a/stop [this]
         (shutdown! app-context)
-        this))))
+        this)
+      (a/restart [this]
+        (run-lifecycle-fns app-context s/stop "stop" (reverse ordered-services))
```

Comment:
What should happen if one of these fail - e.g., if an unhandled exception is thrown up the stack?  Should the process that the apps are running in be shut down?